kernel/src/transaction/mod.rs

```diff
@@ -38,6 +53,27 @@ pub fn add_files_schema() -> &'static SchemaRef {
     &ADD_FILES_SCHEMA
 }
 
+pub(crate) static INTERMEDIATE_ADD_FILES_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    Arc::new(StructType::new(vec![
+        StructField::not_null("path", DataType::STRING),
+        StructField::not_null(
+            "partitionValues",
+            MapType::new(DataType::STRING, DataType::STRING, true),
+        ),
+        StructField::not_null("size", DataType::LONG),
+        StructField::not_null("modificationTime", DataType::LONG),
+        StructField::not_null("dataChange", DataType::BOOLEAN),
+        StructField::nullable(
+            "stats",
+            DataType::struct_type(vec![StructField::nullable("numRecords", DataType::LONG)]),
+        ),
+    ]))
+});
+
+pub fn intermediate_add_files_schema() -> &'static SchemaRef {
```

Comment:
does this need to be `pub`?
```suggestion
fn intermediate_add_files_schema() -> &'static SchemaRef {
```

Reply:
or perhaps just removing entirely and inlining const below?